legend-depot-artifacts-refresh/src/main/java/org/finos/legend/depot/store/artifacts/services/file/FileGenerationHandlerImpl.java

```diff
@@ -134,10 +136,16 @@ public MetadataEventResponse refreshProjectVersionArtifacts(StoreProjectData pro
         return response;
     }
 
+    private String getExtensionKeyFromGeneration(String path, String entityPath)
+    {
+        String type = Arrays.stream(path.replace(PATH_SEPARATOR + entityPath + PATH_SEPARATOR, "").split(PATH_SEPARATOR)).findFirst().get();
```

Comment:
this will throw an exception if empty.
please change the method to return an Optional<String> andn then let deal with the optional in tine 122. we don this for entity path look at teh code in line 111-112